scripts/pgpwave8_upgrade

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+
+usage()
+{
+    cat <<EOF
+usage $0 -i IOCNAME [-d DEVICE] [-r] [-p FIRMWARE_PATH]
+
+Read or upgrade the firmware version of a pgpwave8 given its ioc.
+
+-i The ioc whose pgpwave8's firmware should be upgraded.
+-d The path to the device, defaulting to /dev/datadev_0.
+-r Print the firmware version only. Cannot be used with -p.
+-p The path to the mcs file containing the new firmware.
+Cannot be used with -r. Some firmware images can be found
+in /cds/group/pcds/package/wave8/images. If -r and -p are not provided,
+user will be prompted if they want to use
+/cds/group/pcds/package/wave8/images/latest as the new firmware.
+EOF
+}
+
+remote()
+{
+    #!/bin/bash
+    # shellcheck disable=SC1091
+    source /cds/group/pcds/pyps/conda/rogue/etc/profile.d/conda.sh
+    if ! conda activate rogue_"$ROGUE_VERSION"; then
+        echo "A rogue environment for this version is not available."
+        return 1
+    fi
+    if [[ ! -e $DEVICE ]]; then
+        echo "The device $DEVICE does not exist."
+        return 1
+    fi
+    export PATH=/cds/group/pcds/engineering_tools/latest-released/scripts/:$PATH
```

Comment:
I must have missed this in my last review, having this hard-coded to the `latest-released` directory can cause tons of problems. This means that _all_ versions of this script will use the latest-released area as their path! Old versions, new versions, dev versions, etc.

Most breakingly for dev work, if you edit some other engineering_tools script that is needed here and try to test it, you'll always be running the latest-released version instead.

Instead we need to dynamically find "this release" using the BASH_SOURCE and dirname stuff. There should be examples in other scripts.

Reply:
Thanks for catching this

Reply:
I like your fix for this